quic/s2n-quic-core/src/recovery/rtt_estimator.rs

```diff
@@ -290,6 +294,21 @@ fn abs_difference<T: core::ops::Sub + PartialOrd>(a: T, b: T) -> <T as core::ops
     }
 }
 
+/// Optimized function for adding two durations with a weight
+/// See https://godbolt.org/z/65f9bYEcs
+#[inline]
+fn weighted_add(a: Duration, b: Duration, weight: u64) -> Duration {
```

Comment:
can you add a unit test for this function?

Reply:
https://github.com/aws/s2n-quic/pull/1491/commits/a8fbbc47be25a22ea1adc1d78130132cfab2de3b